src/mpid/ch4/netmod/ofi/ofi_events.c

```diff
@@ -596,6 +596,7 @@ static int am_recv_event(struct fi_cq_tagged_entry *wc, MPIR_Request * rreq)
     /* FI_MULTI_RECV may pack the message at lesser alignment, copy the header
      * when that's the case */
 #define MAX_HDR_SIZE 256        /* need accommodate MPIDI_AMTYPE_RDMA_READ */
+    int need_alignment_check = 0;
```

Comment:
Why `need_alignment_check` is off first? Is it because we checked here already? So it makes sense to introduce and set this flag *after* the alignment check, not before, right?

Even that, it is a bit odd that you set it to off but apparently, it doesn't mean you don't need to do the alignment check. So a quick comment explaining that alignment is already checked would help the readability.

Reply:
oh, I think I need to change the logic.
`need_alignment_check` should be 1 in the beginning, then after repeat, we switch it to 0.

Reply:
Careful, I think you are officially confused 😃